qiskit_machine_learning/algorithms/trainable_model.py

```diff
@@ -202,7 +203,7 @@ def _choose_initial_point(self) -> np.ndarray:
         if self._warm_start and self._fit_result is not None:
             self._initial_point = self._fit_result[0]
         elif self._initial_point is None:
-            self._initial_point = np.random.rand(self._neural_network.num_weights)
+            self._initial_point = algorithm_globals.random.rand(self._neural_network.num_weights)
```

Comment:
```suggestion
            self._initial_point = algorithm_globals.random.random(self._neural_network.num_weights)
```
The deprecated RandomState `rand()` method no longer exists in the Generator replacement. Replacing this by `random()`.